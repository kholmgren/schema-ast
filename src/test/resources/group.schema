definition tenant/role {
	relation member: tenant/user | tenant/group#membership
	permission allowed = member
}

definition tenant/user {}

definition tenant/group {
	relation admin: tenant/user
	relation member: tenant/user
	permission membership = admin + member
}