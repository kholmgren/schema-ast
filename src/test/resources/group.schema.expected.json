{
  "@type" : "schema",
  "definitions" : [ {
    "@type" : "definition",
    "name" : "tenant/role",
    "relations" : [ {
      "@type" : "relation",
      "name" : "member",
      "expression" : {
        "@type" : "or",
        "left" : {
          "@type" : "ref",
          "ref" : "tenant/user"
        },
        "right" : {
          "@type" : "groupRef",
          "ref" : "tenant/group",
          "member" : "membership"
        }
      }
    } ],
    "permissions" : [ {
      "@type" : "permission",
      "name" : "allowed",
      "expression" : {
        "@type" : "ref",
        "ref" : "member"
      }
    } ]
  }, {
    "@type" : "definition",
    "name" : "tenant/user"
  }, {
    "@type" : "definition",
    "name" : "tenant/group",
    "relations" : [ {
      "@type" : "relation",
      "name" : "admin",
      "expression" : {
        "@type" : "ref",
        "ref" : "tenant/user"
      }
    }, {
      "@type" : "relation",
      "name" : "member",
      "expression" : {
        "@type" : "ref",
        "ref" : "tenant/user"
      }
    } ],
    "permissions" : [ {
      "@type" : "permission",
      "name" : "membership",
      "expression" : {
        "@type" : "union",
        "left" : {
          "@type" : "ref",
          "ref" : "admin"
        },
        "right" : {
          "@type" : "ref",
          "ref" : "member"
        }
      }
    } ]
  } ]
}